test_weather_alexander.py

```diff
@@ -0,0 +1,215 @@
+    url = 'http://api.openweathermap.org/data/2.5/weather'
+
+    key = 'f1f0eead8298a901e9069ab5b02dcfdd'  # put your own key here :P
+    params = {'zip': zip_code+','+country_cod,
+              'units': 'metric',
+              'appid': key}
+    r = get(url, params=params)
+    assert r.status_code == 200
+    assert r.reason == 'OK'
+    assert r.elapsed.total_seconds() < 1.0
+    assert r.headers['Content-Type'] == 'application/json; charset=utf-8'
+    assert r.headers['Content-Type'].startswith('application/json')
+
+    data = r.json()
+    # current month:
+    month = time.gmtime().tm_mon
+
+    temp_min, temp_max = year_temp_ranges[zip_code][month]
+
+    assert temp_min < data['main']['temp'] < temp_max
+    print('The temperature is now: ', data['main']['temp'])
+
+
+@pytest.mark.parametrize('coord_lat, coord_lon, point_name', [
+    ('43.6', '39.7303', 'Sochi'),
+])
+def test_weather_geo_coords(coord_lat, coord_lon, point_name):
+    url = 'http://api.openweathermap.org/data/2.5/weather'
+
+    key = 'f1f0eead8298a901e9069ab5b02dcfdd'  # put your own key here :P
+    params = {'lat': coord_lat,
+              'lon': coord_lon,
+              'units': 'metric',
+              'appid': key}
+
+    r = get(url, params=params)
+    assert r.status_code == 200
+    assert r.reason == 'OK'
+    assert r.elapsed.total_seconds() < 1.0
+    assert r.headers['Content-Type'] == 'application/json; charset=utf-8'
+    assert r.headers['Content-Type'].startswith('application/json')
+
+    data = r.json()
+
+    assert data['wind']['speed'] < 10
+    assert data['name'] == point_name
+    print('The temperature is now: ', data['wind']['speed'])
+    print('City name is: ', data['name'])
+
+
+@pytest.mark.parametrize('lon_left, lat_bottom, lon_right, lat_top, zoom , city_in_zone', [
+    ('39', '43', '40', '44', '10', ['Adler', 'Lazarevskoye', 'Sochi']),
+])
+def test_weather_rect_zone(lon_left, lat_bottom, lon_right, lat_top, zoom, city_in_zone):
+    url = 'http://api.openweathermap.org/data/2.5/box/city?'
+
+    key = 'f1f0eead8298a901e9069ab5b02dcfdd'
+    params = {'bbox': lon_left+','+lat_bottom+','+lon_right+','+lat_top+','+zoom,
+              'appid': key}
+
+    r = get(url, params=params)
+
+    assert r.status_code == 200
+    assert r.reason == 'OK'
+    assert r.elapsed.total_seconds() < 1.0
+    assert r.headers['Content-Type'] == 'application/json; charset=utf-8'
+    assert r.headers['Content-Type'].startswith('application/json')
+
+    data = r.json()
+    assert data['list'][0]['name'] == 'Lazarevskoye'
+
+    city_list = []
+    for i in range(len(data['list'])):
+        city_list.append(str(data['list'][i]['name']))
+    assert sorted(city_list) == city_in_zone
+
+
+@pytest.mark.parametrize('city_id, city_name', [
+    ('5809844,491422,484907', ['Seattle', 'Sochi', 'Taganrog']),
+])
+def test_weather_id(city_id, city_name):
+    url = 'http://api.openweathermap.org/data/2.5/group?'
```

Comment:
remove ?

Reply:
@dmitrytokarev I don't understand what you mean, this is a new test with ID endpoint, however, again checked the cities)

Reply:
i meant to remove the question mark sign at the end of the url `?`
The question mark is used to add URL query params, but we add them via `get(url, params=params)`

Reply:
@09041985 please remove that question mark so we can merge